File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/converter/DictionaryConverterImpl.java
package crabster.rudakov.sberschoollesson30hwk.domain.converter;

import java.util.*;

import crabster.rudakov.sberschoollesson30hwk.domain.models.*;
import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.*;

public class DictionaryConverterImpl implements IDictionaryConverter {

    @Override
    public DictionaryItemModel convert(DictionaryItem dictionaryItem) {
        DictionaryItemModel dictionaryItemModel = new DictionaryItemModel();
        dictionaryItemModel.setKeyword(dictionaryItem.getKeyword());
        dictionaryItemModel.setTranslation(dictionaryItem.getTranslation());

        return dictionaryItemModel;
    }

    @Override
    public DictionaryItem reverse(DictionaryItemModel dictionaryItemModel) {
        DictionaryItem dictionaryItem = new DictionaryItem();
        dictionaryItem.setKeyword(dictionaryItemModel.getKeyword());
        dictionaryItem.setTranslation(dictionaryItemModel.getTranslation());

        return dictionaryItem;
    }

    @Override
    public List<DictionaryItem> reverseList(List<DictionaryItemModel> dictionaryItemModels) {
        List<DictionaryItem> result = new ArrayList<>(dictionaryItemModels.size());
        for (DictionaryItemModel model : dictionaryItemModels) {
            result.add(reverse(model));
        }

        return result;
    }

}

File: app/src/main/java/crabster/rudakov/sberschoollesson30hwk/domain/converter/DictionaryConverterImpl.kt
package crabster.rudakov.sberschoollesson30hwk.domain.converter

import crabster.rudakov.sberschoollesson30hwk.domain.models.DictionaryItem
import crabster.rudakov.sberschoollesson30hwk.domain.repositories.models.DictionaryItemModel

/**
 * Converter for model classes [DictionaryItem] and [DictionaryItemModel]
 */
class DictionaryConverterImpl : IDictionaryConverter {

    override fun convert(dictionaryItem: DictionaryItem) =
        DictionaryItemModel(
            keyword = dictionaryItem.keyword,
            translation = dictionaryItem.translation,
            logo = dictionaryItem.logo
        )


    override fun reverse(dictionaryItemModel: DictionaryItemModel) =
        DictionaryItem(
            keyword = dictionaryItemModel.keyword,
            translation = dictionaryItemModel.translation,
            logo = dictionaryItemModel.logo
        )

    override fun reverseList(dictionaryItemModels: List<DictionaryItemModel>) =
        dictionaryItemModels.map { reverse(it) }.toList()

}